Zend/zend_API.c

```diff
@@ -1323,9 +1339,13 @@ ZEND_API HashTable *zend_separate_class_constants_table(zend_class_entry *class_
 
 	ZEND_HASH_FOREACH_STR_KEY_PTR(&class_type->constants_table, key, c) {
 		if (Z_TYPE(c->value) == IS_CONSTANT_AST) {
-			new_c = zend_arena_alloc(&CG(arena), sizeof(zend_class_constant));
-			memcpy(new_c, c, sizeof(zend_class_constant));
-			c = new_c;
+			if (c->ce == class_type) {
+				new_c = zend_arena_alloc(&CG(arena), sizeof(zend_class_constant));
+				memcpy(new_c, c, sizeof(zend_class_constant));
+				c = new_c;
+			} else {
+				c = inherit_class_constant(class_type, key);
```

Comment:
Can we use ``c = zend_hash_find_ptr(CE_CONSTANTS_TABLE(c->ce), name);`` ?

Reply:
Yes, using `CE_CONSTANTS_TABLE(c->ce)` makes much more sense.